test/ArcxAnalyticsSdk.test.ts

```diff
@@ -89,290 +110,460 @@ describe('(unit) ArcxAnalyticsSdk', () => {
+        }
+        await analyticsSdk.connectWallet(testAttributes)
+        expect(eventStub).calledOnceWithExactly(CONNECT_EVENT, testAttributes)
+      })
+    })
 
-      await analyticsSdk.referrer('https://test.site')
-      expect(eventStub.calledOnceWith(REFERRER_EVENT, { referrer: 'https://test.site' })).to.be.true
+    describe('#transaction', () => {
+      it('calls event() with the given attributes', async () => {
+        const eventStub = sinon.stub(analyticsSdk, 'event')
+        const testAttributes = {
+          chain: '1',
+          transactionHash: '0x123456789',
+          metadata: { timestamp: '123456' },
+        }
+        await analyticsSdk.transaction(testAttributes)
+        expect(eventStub).calledOnceWithExactly(TRANSACTION_EVENT, {
+          chain: '1',
+          transaction_hash: '0x123456789',
+          metadata: { timestamp: '123456' },
+        })
+      })
     })
 
-    describe('#transaction', async () => {
-      const attributes = {
-        chain: '1',
-        transactionHash: '0x123456789',
-        metadata: { timestamp: '123456' },
-      }
+    describe('#referrer', () => {
+      it('calls event() with the given attributes', async () => {
+        const eventStub = sinon.stub(analyticsSdk, 'event')
+        await analyticsSdk.referrer(TEST_REFERRER)
+        expect(eventStub.calledOnceWith(REFERRER_EVENT, { referrer: TEST_REFERRER })).to.be.true
+      })
+    })
+  })
 
+  describe('Private functions', () => {
+    describe('#_trackFirstPageVisit', () => {
       let eventStub: sinon.SinonStub
 
       beforeEach(() => {
         eventStub = sinon.stub(analyticsSdk, 'event')
       })
 
-      it('all parameters are passed', async () => {
-        await analyticsSdk.transaction(attributes)
+      it('sets the current window location to sessionStorage if trackPages is true', () => {
+        analyticsSdk['sdkConfig'].trackPages = true
 
-        expect(
-          eventStub.calledOnceWith(TRANSACTION_EVENT, {
-            chain: attributes.chain,
-            transaction_hash: attributes.transactionHash,
-            metadata: attributes.metadata,
-          }),
-        ).to.be.true
+        expect(sessionStorage.getItem(CURRENT_URL_KEY)).to.be.null
+        analyticsSdk['_trackFirstPageVisit']()
+
+        expect(sessionStorage.getItem(CURRENT_URL_KEY)).to.eq(TEST_JSDOM_URL)
+        analyticsSdk['sdkConfig'].trackPages = false
       })
-    })
 
-    it('super nested attributes are supported', async () => {
-      const pageAttributes = { layer1: { layer2: { layer3: { layer4: 'hello!' } } } }
-      const eventStub = sinon.stub(analyticsSdk, 'event')
+      it('emits a FIRST_PAGE_VISIT event with url if trackPages is true', () => {
+        analyticsSdk['sdkConfig'].trackPages = true
 
-      await analyticsSdk.event('TEST_EVENT', pageAttributes)
+        analyticsSdk['_trackFirstPageVisit']()
+        expect(eventStub).calledOnceWithExactly(FIRST_PAGE_VISIT, { url: TEST_JSDOM_URL })
```

Comment:
we can just check if `_trackFirstPageVisit` is called. But event is good as well